cardano-api/src/Cardano/Api/Protocol/Types.hs

```diff
@@ -138,3 +173,14 @@ data BlockType blk where
 deriving instance Eq (BlockType blk)
 deriving instance Show (BlockType blk)
 
+reflBlockType :: BlockType blk -> BlockType blk' -> Maybe (blk :~: blk')
```

Comment:
Perhaps instead define an instance of the `base` class for this: https://hackage.haskell.org/package/base-4.16.1.0/docs/Data-Type-Equality.html#t:TestEquality